### PR TITLE
doc: services: pm: Reword System PM Doc

### PR DESCRIPTION
**Pull request by decsny**
I was reading through the system PM documentation and found it slightly hard to read. So, I made some changes that I felt improved this page.

The following changes:
- Generally reword all the paragraphs to hopefully by clearer and easier to read by using more straighforward and direct wording, and avoiding run on sentences and ambiguous/wrong grammar, etc.
- Move Examples to the bottom of the page.
- Fix a typo in the code expression where there was an extra parentheses.

### CODE REVIEW

File: doc/services/pm/system.rst [hunk 1]
```diff
@@ -3,42 +3,51 @@
...
-have higher wake latencies.
+* ACTIVE: System is fully powered and active.
+* RUNTIME_IDLE: Cores enter deepest idle state waiting for interrupts,
+  all other devices are unaffected.
+* SUSPEND_TO_IDLE: May put some peripherals into low power states.
+* STANDBY: Peripherals are put into low power states, and non boot cores are powered off.
+* SUSPEND_TO_RAM: Powers off as much of the system as possible.
+  Execution state and device state is held in ram.
+* SUSPEND_TO_DISK: saves to a disk instead of ram.
+* SOFT_OFF: essentially nothing is preserved, wake-up will be like a system reboot.
```
Comment (ceolin) [inline doc/services/pm/system.rst:47]: I am not a big fan of duplicate this information. Can we just link to their definition ?

File: doc/services/pm/images/system-pm.svg
Comment (ceolin) [inline doc/services/pm/images/system-pm.svg:1]: Please don't bother changing it. The diagram is outdated, there is no low power states for devices. I've it addressed in a different pr.

**Review by ceolin — CHANGES_REQUESTED** [review]
Very nice changes. Minor comments.

File: doc/services/pm/system.rst (hunk 1 above)
Comment (decsny) [reply doc/services/pm/system.rst:47]: Fair enough

File: doc/services/pm/system.rst [hunk 2]
```diff
@@ -3,42 +3,51 @@
...
-source.
+The kernel enters the idle state when it has nothing to schedule.
+Enabling :kconfig:option:`CONFIG_PM` allows the kernel to call upon the
+power management subsystem to put an idle system into one of the supported power states.
+
+The PM subsystem selects the appropriate power state to transition to
+depending on the following factors:
+
+* The amount of time prescribed by the kernel to suspend upon invocation of the power manager.
+* The chosen power management policy.
```
[the remark this answers is not in the document]
Comment (decsny) [reply doc/services/pm/system.rst:14]: Okay. This is what I was trying to say here. See the previous wording was "based
on the selected power management policy and the duration of the idle time
allotted by the kernel." , I just turned these into bullet points. So how do we make the wording more clear if this new version is also still not clear?

**Review by dleach02 — APPROVED** [review]

**Comment by decsny** [comment]
@ceolin ping

**Review by ceolin — APPROVED** [review]
Thanks for the changes !

**Pull request merged**